pkg/compactor/compactor.go

```diff
@@ -607,6 +613,8 @@ func (c *Compactor) compactUsers(ctx context.Context) {
 		if succeeded && compactionErrorCount == 0 {
 			c.compactionRunsCompleted.Inc()
 			c.compactionRunsLastSuccess.SetToCurrentTime()
+		} else if interrupted {
+			c.compactionRunsInterrupted.Inc()
 		} else {
 			c.compactionRunsFailed.Inc()
```

Comment:
If there is failure before the interruption, we will only go L617 and miss `c.compactionRunsFailed` metric.
Probably increament the failure metric in a separate if block.

Reply:
> If there is a failure before the interruption

In practice, this cannot currently happen. But I agree what you're saying makes sense in this local part of the code.  Will make the change.

Reply:
Yes, that sounds good!

Reply:
Cool, makes sense to me too.  Will write additional tests as well.